modules/doc/content/getting_started/installation/inl_hpc_install_moose.md

```diff
@@ -1,71 +1,109 @@
+## Containerization
 
-!style! halign=left
-While operating on one of our [!ac](INL) [!ac](HPC) clusters, you need only load a couple of
-modules to obtain a proper developer environment:
-!style-end!
+The development environment is deployed using containerization, which is a deployment
+practice that bundles all application (i.e., MOOSE or a MOOSE-based app) dependencies need to run
+on any host. This means that applications built within this environment can be executed
+across all INL HPC clusters. That is, you can build an application within a compute node on
+Sawtooth at INL and execute the same application (providing you follow the instructions
+that follow) on both Bitterroot and Lemhi, in parallel.
 
-- +[!ac](HPC) Sawtooth or Lemhi+ (required each time you log in):
+## Versioning id=versioning
 
-  ```bash
-  module load use.moose moose-dev
-  ```
+The version of the dependencies used within MOOSE (PETSc, libMesh, etc) are tied directly
+to a given version of MOOSE. Given a version of MOOSE (typically a Git commit hash), we produce
+a version of the development environment that contains the correct version of these dependencies.
+Whenever you are compiling MOOSE or a MOOSE-based application, you should use the corresponding
+version of the development environment.
 
-If you would prefer not having to perform the above step each time you log in, you can append the
-above command to your shell initialization file:
+In order to determine the version of the development environment to use when building, run the
+following within the MOOSE directory:
 
 ```bash
-echo "module load use.moose moose-dev" >> ~/.bash_profile
+./scripts/versioner.py moose-dev
 ```
 
-## Cloning MOOSE
+where the output will be a hashed version:
 
-!style! halign=left
-Follow the below instructions, and replace `cluster_name` placeholder with either `lemhi` or
-`sawtooth` accordingly.
-!style-end!
+!versioner! code
+__VERSIONER_VERSION_MOOSE_DEV__
+!versioner-end!
 
-!alert! warning title=Replace +'cluster_name'+ accordingly!
-[!ac](INL) [!ac](HPC) machines use a shared home directory structure (things you create/do on Lemhi
-will be available on Sawtooth). If you do not separate your projects directory based on the cluster
-you are operating on (or some other identifier), you risk developing under one environment and then
-executing on another. Mistakes like this will cause the odd failure and cost you time to solve/fix.
-!alert-end!
+In this case, the version of `moose-dev` that should be used is [!versioner!version package=moose-dev].
+For most applications in which a MOOSE submodule is used, you will commonly excute:
 
-!style! halign=left
-!template load file=installation/clone_moose.md.template PATH=~/cluster_name/projects
-!style-end!
+!versioner! code
+./moose/scripts/versioner.py moose-dev
+!versioner-end!
 
-## Build PETSc and libMesh
+Note that this version will change in time as MOOSE changes, or if you use an older version of MOOSE.
```

Comment:
I would add explanation on what is the default version of `moose-dev-openmpi`. I guess it is safe if the default version can compile an application without an issue since moose has been tested against applications. I feel asking users to specify the version number can be an advanced feature.

Reply:
> I guess it is safe if the default version can compile an application without an issue since moose has been tested against applications

This isn't necessarily the case, which is why I'm so explicit that you should use versions here. I don't want to suggest to users that they should use the default, because when we change the default (happens weekly), it will likely break their install. Just like updating conda would.

I do agree that including information on openmpi is good idea. I plan on going into this further in a later PR. Including a section on mpich